src/hnsw/options.c

```diff
@@ -91,15 +95,28 @@ usearch_metric_kind_t ldb_HnswGetMetricKind(Relation index)
 
     if(fnaddr == l2sq_dist || fnaddr == vector_l2sq_dist) {
         return usearch_metric_l2sq_k;
-    } else if(fnaddr == hamming_dist) {
+    } else if(fnaddr == hamming_dist || fnaddr == vector_hamming_dist) {
         return usearch_metric_hamming_k;
-    } else if(fnaddr == cos_dist) {
+    } else if(fnaddr == cos_dist || fnaddr == vector_cos_dist) {
         return usearch_metric_cos_k;
     } else {
         elog(ERROR, "could not find distance function for index");
     }
 }
 
+void pgvector_compat_assign_hook(bool status, void *extra)
+{
+    LDB_UNUSED(extra);
+
+    if(status) {
+        post_parse_analyze_hook = original_post_parse_analyze_hook;
+        ExecutorStart_hook = original_ExecutorStart_hook;
+    } else {
+        post_parse_analyze_hook = post_parse_analyze_hook_with_operator_check;
```

Comment:
Yes, I updated this part to store current hooks as originals, but I didn't assign NULL in the branch above, as the extension might be unloaded while we assign nulls to them and it will assign NULL to the hooks here in PG_fini:
```
void _PG_fini(void)
{
    // Return back the original hook value.
    post_parse_analyze_hook = original_post_parse_analyze_hook;
    ExecutorStart_hook = original_ExecutorStart_hook;
}
```

Reply:
I have added check in `_PG_fini` to reset the hooks only if they are set to our hooks, and set the original hooks to NULL in the branch above. https://github.com/lanterndata/lantern/pull/240/commits/5eab1302bcd95a9944e9d3255cc8dcaf6a45db35